packages/mobile/src/identity/pgpnp.ts

```diff
@@ -0,0 +1,50 @@
+// Utilities for interacting with the PGPNP service
+// PGPNP == Pretty Good Phone Number Privacy
+
+import { ContractKit } from '@celo/contractkit'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import networkConfig from 'src/geth/networkConfig'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'identity/privacy'
+
+export async function postToPGPNP<ResponseType>(
+  account: string,
+  contractKit: ContractKit,
+  body: object,
+  endpoint: string
+) {
+  Logger.debug(`${TAG}@postToPGPNP`, `Posting to ${endpoint}`)
+
+  // Sign payload using account privkey
+  const bodyString = JSON.stringify(body)
+  const authHeader = await contractKit.web3.eth.sign(bodyString, account)
```

Comment:
@yorhodes will this require any work with your PR?

Reply:
the provider will intercept this already